replica/backup.go

```diff
@@ -87,7 +90,8 @@ func (rb *Backup) ReadSnapshot(snapID, volumeID string, start int64, data []byte
 	if err := rb.assertOpen(id, volumeID); err != nil {
 		return err
 	}
-
+	rb.lock.Lock()
```

Comment:
Why is the lock not covering `rb.assertOpen`?

Reply:
Earlier I had locks inside assertOpen and I had made this change thinking that I should not have a lock inside a lock. Later I removed locks inside assertOpen as it will be eventually removed from shuo's commit. 

As you pointed out, for this commit, we need the lock to cover rb.assertOpen. Will fix this